sources/cqlrt_common.c

```diff
@@ -4214,49 +4224,55 @@ cql_code cql_rebuild_recreate_group(sqlite3 *_Nonnull db, cql_string_ref _Nonnul
 
   cql_code rc = SQLITE_OK;
   // Execute all delete table drops
-  for (cql_int32 i = 0; i < cql_string_list_get_count(deleteList); i++){
+  for (cql_int32 i = 0; i < cql_string_list_get_count(deleteList); i++) {
     cql_string_ref delete = cql_string_list_get_string(deleteList, i);
     rc = cql_exec_internal(db, delete);
     if (rc != SQLITE_OK) goto cleanup;
   }
   // Execute all table drops based on the list of creates given by the CQL
   // upgrader backwards.
   // Intuitively, need to drop the tables with the most dependencies first.
-  for (cql_int32 i = cql_string_list_get_count(tableList) - 1; i >= 0; i--){
+  for (cql_int32 i = cql_string_list_get_count(tableList) ; --i >= 0; ) {
     cql_string_ref tableCreate = cql_string_list_get_string(tableList, i);
-    char* table_name = _cql_create_table_name_from_table_creation_statement(tableCreate);
-    cql_int32 bytes = (cql_int32)(strlen(table_name)) + sizeof("DROP TABLE IF EXISTS ");
-    char* drop = malloc(bytes);
-    snprintf(drop, bytes, "DROP TABLE IF EXISTS %s", table_name);
-    rc = cql_exec(db, drop);
+    char *table_name = _cql_create_table_name_from_table_creation_statement(tableCreate);
+
+    cql_bytebuf drop;
```

Comment:
This is the only "real" change.  This is just a lot less error prone and it's the idiomatic way we build buffers in the cqlrt